scene/resources/animation.cpp

```diff
@@ -5476,471 +5476,618 @@ bool Animation::_fetch_compressed_by_index(uint32_t p_compressed_track, int p_in
 }
 
 // Helper math functions for Variant.
+bool Animation::is_variant_interpolatable(const Variant p_value) {
+	Variant::Type type = p_value.get_type();
+	return (type >= Variant::BOOL && type <= Variant::STRING_NAME) || type == Variant::ARRAY || type >= Variant::PACKED_INT32_ARRAY; // PackedByte is unsigned, so it would be better to ignore since blendigng uses float.
+}
+
+Variant Animation::cast_to_blendwise(const Variant p_value) {
+	switch (p_value.get_type()) {
+		case Variant::BOOL:
+		case Variant::INT: {
+			return p_value.operator real_t();
+		} break;
+		case Variant::STRING:
+		case Variant::STRING_NAME: {
+			return string_to_array(p_value);
+		} break;
+		case Variant::RECT2I: {
+			return p_value.operator Rect2();
+		} break;
+		case Variant::VECTOR2I: {
+			return p_value.operator Vector2();
+		} break;
+		case Variant::VECTOR3I: {
+			return p_value.operator Vector3();
+		} break;
+		case Variant::VECTOR4I: {
+			return p_value.operator Vector4();
+		} break;
+		case Variant::PACKED_INT32_ARRAY: {
+			return p_value.operator PackedFloat32Array();
+		} break;
+		case Variant::PACKED_INT64_ARRAY: {
+			return p_value.operator PackedFloat64Array();
+		} break;
+		default: {
+		} break;
+	}
+	return p_value;
+}
+
+Variant Animation::cast_from_blendwise(const Variant p_value, const Variant::Type p_type) {
+	switch (p_type) {
+		case Variant::BOOL: {
+			return p_value.operator real_t() >= 0.5;
+		} break;
+		case Variant::INT: {
+			return (int)Math::round(p_value.operator real_t());
+		} break;
+		case Variant::STRING: {
+			return array_to_string(p_value);
+		} break;
+		case Variant::STRING_NAME: {
+			return StringName(array_to_string(p_value));
+		} break;
+		case Variant::RECT2I: {
+			Rect2 old_val = p_value.operator Rect2();
+			return Rect2i(old_val.round());
+		} break;
+		case Variant::VECTOR2I: {
+			Vector2 old_val = p_value.operator Vector2();
+			return Vector2i(old_val.round());
+		} break;
+		case Variant::VECTOR3I: {
+			Vector3 old_val = p_value.operator Vector3();
+			return Vector3i(old_val.round());
+		} break;
+		case Variant::VECTOR4I: {
+			Vector4 old_val = p_value.operator Vector4();
+			return Vector4i(old_val.round());
+		} break;
+		case Variant::PACKED_INT32_ARRAY: {
+			PackedFloat32Array old_val = p_value.operator PackedFloat32Array();
+			PackedInt32Array new_val;
+			for (int i = 0; i < old_val.size(); i++) {
+				new_val.push_back((int32_t)Math::round(old_val[i]));
+			}
```

Comment:
```suggestion
			PackedInt32Array new_val;
			new_val.resize(old_val.size());
			int *new_val_w = new_val.ptrw();
			for (int i = 0; i < old_val.size(); i++) {
				new_val_w[i] = (int32_t)Math::round(old_val[i]);
			}
```
More efficient. Same below.